Selection not in cache after partial

When copying subsequent partial selection, the selection just before is removed from cache.
I found something in `clipmenud` that might be a bug, here is a possible fix:

```diff
diff --git a/clipmenud b/clipmenud
index 3c4d27b252..4eba9acfd3 100755
--- a/clipmenud
+++ b/clipmenud
@@ -181,9 +181,6 @@ while true; do
             rm -- "${last_filename[$selection]}"
         fi
 
-        last_data[$selection]=$data
-        last_filename[$selection]=$filename
-
         first_line=$(get_first_line "$data")
 
         debug "New clipboard entry on $selection selection: \"$first_line\""
@@ -191,6 +188,9 @@ while true; do
         cache_file_output="$(date +%s%N) $first_line"
         filename="$cache_dir/$(cksum <<< "$first_line")"
 
+        last_data[$selection]=$data
+        last_filename[$selection]=$filename
+
         # Recover without restart if we deleted the entire clip dir.
         # It's ok that this only applies to the final directory.
         # shellcheck disable=SC2174
```

--- a/clipmenud
+++ b/clipmenud
@@ -181,15 +181,15 @@ while true; do
             rm -- "${last_filename[$selection]}"
         fi
 
-        last_data[$selection]=$data
-        last_filename[$selection]=$filename
-
         first_line=$(get_first_line "$data")
 
         debug "New clipboard entry on $selection selection: \"$first_line\""
 
         cache_file_output="$(date +%s%N) $first_line"
         filename="$cache_dir/$(cksum <<< "$first_line")"
+
+        last_data[$selection]=$data
+        last_filename[$selection]=$filename
 
         # Recover without restart if we deleted the entire clip dir.
         # It's ok that this only applies to the final directory.

